embassy-usb-synopsys-otg/src/lib.rs

```diff
@@ -30,12 +30,20 @@ pub unsafe fn on_interrupt<const MAX_EP_COUNT: usize>(r: Otg, state: &State<MAX_
     trace!("irq");
 
     let ints = r.gintsts().read();
-    if ints.wkupint() || ints.usbsusp() || ints.usbrst() || ints.enumdne() || ints.otgint() || ints.srqint() {
+    if ints.wkupint()
+        || ints.usbsusp()
+        || ints.usbrst()
+        || ints.enumdne()
+        || ints.otgint()
+        || ints.srqint()
+        || ints.sof()
+    {
         // Mask interrupts and notify `Bus` to process them
         r.gintmsk().write(|w| {
             w.set_iepint(true);
             w.set_oepint(true);
             w.set_rxflvlm(true);
+            w.set_sofm(true);
```

Comment:
you should probably make this optional. One interrupt every 125 us is quite a lot.

Reply:
sorry for the delay. Why don't you add a `enable_sof_interrupts` to the `Config` struct passed to the `Driver` impl?
Then just make sure to update the docs to reflect the idea that it will cause a lot of interrupts and should only be enabled if you really it.